main.tf

```diff
@@ -1 +1,62 @@
-# Your Terraform code goes here :)
+locals {
+  policy_name = length(var.policy_name) > 0 ? var.policy_name : "${var.base_name}_policy"
+}
+
+module "lacework_oci_credentials" {
+  source        = "lacework/iam-user/oci"
+  version       = ">= 0.2.0"
+  tenancy_id    = var.tenancy_id
+  create        = var.create
+  freeform_tags = var.freeform_tags
+  email         = var.user_email
+  base_name     = var.base_name
+  user_name     = var.user_name
+  group_name    = var.group_name
+}
+
+resource "oci_identity_policy" "lacework_policy" {
+  count          = var.create ? 1 : 0
+  compartment_id = var.tenancy_id
+  description    = "Permission policy for the Lacework CSPM user"
+  name           = local.policy_name
+  freeform_tags  = var.freeform_tags
+  statements = [
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect compartments in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to read buckets in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect volumes in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect security-lists in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to read users in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect groups in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect policies in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect domains in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect tag-defaults in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect instances in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect subnets in tenancy",
+    "Allow group '${module.lacework_oci_credentials.group_name}' to inspect network-security-groups in tenancy",
+  ]
```

Comment:
@fdong-lw @akshara-lw Can you take a look at this set of permissions needed for Lacework configuration integration. Is this right as a minimal set of permissions?

Reply:
those are correct and verified for preview preview. It will cover 15 APIs. In future, we will add more permissions for new coverage. The list is tracked at https://docs.google.com/spreadsheets/d/1PLVmmE3yy7WIo5U3rxh36mN7A6bBYtzVPzs6g8m6o0I/edit?pli=1#gid=1516278081 We can start with this, but keep in mind this list would be changed time to time, so think about how to manage it. maybe extract the values to any template/config file?

Reply:
Thanks for taking a look. I don't think there's any way around updating this list as we need permissions if we want to maintain least-privilege. I can extract this to a Terraform template file if you think that will simplify things. Let me get on that.

Reply:
I looked into it, and it does not seem like extracting this to a template file will make things more readable. I think the best bet is to simply update this list when we need more permissions.